pkg/chainsync/chainsync.go

```diff
@@ -79,15 +79,21 @@ func New(s p2p.Streamer, backend transaction.Backend) (*ChainSync, error) {
 
 // Prove asks a peer to prove they know a certain block height on the
 // current used eth backend.
-func (s *ChainSync) Prove(ctx context.Context, peer swarm.Address, blockheight uint64) ([]byte, error) {
+func (s *ChainSync) Prove(ctx context.Context, peer swarm.Address, blockheight uint64) (hash []byte, err error) {
 	if !s.outLimiter.Allow(peer.ByteString(), 1) {
 		return nil, errRateLimitExceeded
 	}
 	stream, err := s.streamer.NewStream(ctx, peer, nil, protocolName, protocolVersion, syncStreamName)
 	if err != nil {
 		return nil, fmt.Errorf("new stream: %w", err)
 	}
-	defer stream.Close()
+	defer func() {
+		if err != nil {
+			_ = stream.Reset()
```

Comment:
To the best of my knowledge, when a read/write fro/to a stream returns an error there is no need to reset the stream, since it is already in an errored state, therefore the reset is ineffectual. This is why the original `Close` suffices.